src/transport/Session.h

```diff
@@ -300,10 +300,10 @@ class Session
 #if INET_CONFIG_ENABLE_TCP_ENDPOINT
     // This API is used to associate the connection with the session when the
     // latter is about to be marked active. It is also used to reset the
-    // connection to a nullptr when the connection is lost and the session
-    // is marked as Defunct.
-    ActiveTCPConnectionState * GetTCPConnection() const { return mTCPConnection; }
-    void SetTCPConnection(ActiveTCPConnectionState * conn) { mTCPConnection = conn; }
+    // connection when the connection is lost and the session is marked as Defunct.
+    inline const ActiveTCPConnectionHolder & GetTCPConnection() const { return const_cast<Session *>(this)->mTCPConnection; }
```

Comment:
Why do you need the const_cast here?  You're returning a const ref....

Reply:
Fixed in follow-up